cloudwatch/cloudwatch.go

```diff
@@ -645,6 +645,22 @@ func (output *OutputPlugin) putLogEvents(stream *logStream) error {
 				stream.nextSequenceToken = &parts[len(parts)-1]
 
 				return output.putLogEvents(stream)
+			} else if awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
+				// a log group or a log stream should be re-created after it is deleted and then retry
+				logrus.Errorf("[cloudwatch %d] Encountered error %v; detailed information: %s\n", output.PluginInstanceID, awsErr, awsErr.Message())
+				if strings.Contains(awsErr.Message(), "group") {
+					if err := output.createLogGroup(&Event{group: stream.logGroupName}); err != nil {
+						logrus.Error(err)
+						return err
+					}
+				} else if strings.Contains(awsErr.Message(), "stream") {
+					if _, err := output.createStream(&Event{group: stream.logGroupName, stream: stream.logStreamName}); err != nil {
+						logrus.Error(err)
+						return err
+					}
+				}
+
+				return fmt.Errorf("[cloudwatch %d] A Log group/stream did not exist, re-created it. Will retry PutLogEvents on next flush", output.PluginInstanceID)
```

Comment:
The above two `logrus.Error` lines do not have the `[cloudwatch %d]` identifier. The error message on this line is formatted like a log line instead of an error message. This error message is also not printed here, only returned. I recommend some cleanup among the error handling in this procedure.

Reply:
Good point, I don't think we ever include the plugin ID in errors, just in log messages.

Reply:
Thanks for the suggestion and I will do some cleanup on it